documentation/docs/user_guide/tutorial/compilation.md

```diff
@@ -1,2 +1,89 @@
+
+CABLE supports both serial and parallel applications. 
+
+???+ tip "Serial for single-site"
+    On a single processor, for single-site investigations with a suitable restart file, CABLE usually executes a year of forcing data in a matter of seconds/minutes. Even in the absence of a restart file, thus requiring spinup of the model, this usually doesn't take more than a few minutes. In this case a serial version of CABLE will suffice. 
+
+???+ tip "Multiprocessor for global simulations"
+    For global (or regional) offline simulations, CABLE can still be run in serial mode (about 15 minutes/year for GSWP global run at 1x1 degree resolution). However, running on multiple processors speeds up the simulation considerably.
+
+CABLE can be built using the BASH script [build3.sh][build3] in the **offline/** directory. 
+
+???+ tip "Build on other HPC"
+    The function *host_gadi* in the build script shows the appropriate configuration to build on Gadi.
+    This may be of use as a template building CABLE on another Linux/HPC system.
+
+Both the serial and parallel executable for CABLE offline are built in the `offline/` directory by executing 
+the build script. 
+
+To build the serial model execute: 
+
+    ./build3.sh
+
+To build the parrallel model execute the same build script but with the argumnent **mpi**.   
+
+    ./build3.sh mpi
+
+
+The build script:
+
+1. loads modules for the Fortran compiler and the netCDF library
+2. sets the compiler flags
+3. creates a hidden tempoary directory (`.tmp/`) in which it then compiles CABLE.  
+   
+A [Makefile][makefile] compiles the CABLE code that is common to both serial and parallel CABLE. This includes all the files under the `science/`, `util/`, and `params/` directories. 
+
+A serial compilation then calls `serial_cable` to compile the serial driver and link everything together to produce the executable, `cable`.
+
+A parallel compilation then calls `parallel_cable` to compile the required MPI driver(s) and link everything together to produce the executable, `cable-mpi`.
+
+The hidden `.tmp/` directory is really only needed for technical reasons and you should never need to go 
+into the `.tmp/` directory. However, if for some reason you want the **.o** object files created by the compiler, they persist in this directory. Alternatively, it is sometimes useful to verify that the files you want to compile are actually making it into this directory. This is particuarly relevant if you are adding new files to CABLE. 
+
+???+ warning "Do not change files under `.tmp/`"
+    If you change the files in `.tmp/` directly, the next build will not pick up these changes and will overwrite them.
+
+One of the features of the build process is that only source files which are
+modified are re-built, followed by their dependents. This is possible because the .''tmp'' directory is
+overwritten by the build script, preserving timestamps of the source files from their original location.
```

Comment:
I believe this is currently broken and one needs to remove the .tmp/ directory each time and reset the compilation to start from zero. Is that really so?

Reply:
hmmm not sure what you mean. I can build, and continually rebuild and it only recompiles the file I'm editing it also rebuilds files that depend on it so they pick up the changes).  I only rm -fr .tmp etc when I move between mph and ifort. I actually do it at critical junctures as well just to be ultra cautious.

Reply:
OK. That's what I thought. It's only needed if changing between serial and parallel compilation. I'll add a warning about this in the doc.